sunpy/util/util.py

```diff
@@ -212,3 +212,35 @@ def expand_list_generator(inp):
                 yield nested_item
         else:
             yield item
+
+def partial_match(key, dictionary):
+    """
+    Return value/values based on the partial key. Each element of the partial key is matched against the keys of the
+    dictionary and if a match is found the value of the key is returned. Even a partial match works here i.e 
+    even if the key matches partially a value is returned. Note that this function uses a generator so we need to 
+    type cast the output to a list to get the values.
+    For eg: If we have a dictionary like
+    `d = {('abc','def','ghi') : 1, ('abc', 'def', 'xyz') : 2, ('pqr', 'lmn', 'tuv') : 3}` then a command
+    `key = list(partial_match(('abc', 'def', None), d))` will return a list containing the values `[1,2]` as the given
+    key partially matches with two keys i.e ('abc','def','ghi') and ('abc','def','xyz') in the dictionary.
```

Comment:
This should be in an example section and not here.

Reply:
Can we add description in the example section? I mean can we describe how things work in the example section?

Reply:
I mean you can describe it here and have this code actually run in an example section.